src/testing/utils.tsx

```diff
@@ -102,21 +102,36 @@ export const submitFormikForm = (
   }
 };
 
-type WrapperProps = { state?: RootState };
+type WrapperProps = {
+  parentRoute?: string;
+  routePattern?: string;
+  state?: RootState;
+};
 
 const BrowserRouterWithProvider = ({
   children,
+  parentRoute,
+  routePattern,
   state,
 }: WrapperProps & { children: React.ReactElement }) => {
   const getMockStore = (state: RootState) => {
     const mockStore = configureStore();
     return mockStore(state);
   };
 
+  const route = <Route element={children} path={routePattern} />;
   return (
     <Provider store={getMockStore(state || rootStateFactory())}>
       <BrowserRouter>
-        <CompatRouter>{children}</CompatRouter>
+        <CompatRouter>
+          {routePattern ? (
+            <Routes>
+              {parentRoute ? <Route path={parentRoute}>{route}</Route> : route}
```

Comment:
Components need to be rendered inside a Route for some hooks to work e.g. `useParams`.